src/addon-manager.js

```diff
@@ -760,12 +760,14 @@ class AddonManager extends EventEmitter {
       this.adapters.delete(a.id);
     }
 
-    const plugin = this.getPlugin(packageName);
-    if (plugin) {
-      plugin.kill();
-    }
+    const cleanup = () => {
+      const plugin = this.getPlugin(packageName);
+      if (plugin) {
+        plugin.kill();
+      }
+    };
 
-    return Promise.all(unloadPromises);
+    return Promise.all(unloadPromises).then(() => cleanup(), () => cleanup());
```

Comment:
I realized while I was shoveling the driveway that we also reuse the Plugin object.
https://github.com/mozilla-iot/gateway/blob/master/src/addons/plugin/plugin-server.js#L99

We should probably create a new one and replace the old one. That way if the old one still has timers etc they'll still be valid and we won't wind up killing the new process we just launched